src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -269,6 +286,10 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         traceInfoIf(isCI, `Clear controller mapping for ${document.uri.toString()}`);
         const loadControllersPromise = this.loadNotebookControllers();
 
+        if (isPythonNotebook(getNotebookMetadata(document)) && this.extensionChecker.isPythonExtensionInstalled) {
+            // If we know we're dealing with a Python notebook, load the active interpreter as a kernel asap.
+            this.createActiveInterpreterController().catch(noop);
```

Comment:
Does this need to be dedupified later? Won't it be returned by the code after this again?

Reply:
Agreed, will add a simple flag to ensure we don't do this again.

Reply:
Fixed.